### PR TITLE
feat(kit): new sizes for `ProgressBar`

### PR DESCRIPTION
**Pull request by 40oleg**
New attribute has name "compact" and equals to 16px

https://github.com/Tinkoff/taiga-ui/issues/4058
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behavior?
[Add extra sizing options for progress bar# ](https://github.com/Tinkoff/taiga-ui/issues/4058)(https://github.com/Tinkoff/taiga-ui/issues/4058)

## What is the new behavior?
Now users can use additional size for the progress component with new attribute compact. It has 1rem height in css. (s-size=0.5rem, m-size=1.5rem)

Also added some description in API section for users know that there is size between "s" and "m".
I can not add new l-size because of the breaking changes.

### CODE REVIEW

File: projects/kit/components/progress/progress-bar/progress-bar.component.less [hunk 1]
```diff
@@ -72,6 +72,10 @@
         height: 0.5rem;
     }
 
+    &[compact] {
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less:75]: @waterplea @splincode @vladimirpotekhin what do you think about it ?

[the remark this answers is not in the document]
Comment (40oleg) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less]: @nsbarsukov 
I rewrote it as you and Alex proposed.
The solution based on similar problem in Avatar component.

[the remark this answers is not in the document]
Comment (evantrimboli) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less]: FYI: https://github.com/Tinkoff/taiga-ui/issues/4058

This issue is related

[the remark this answers is not in the document]
Comment (40oleg) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less]: It seems that I have a problem with demo.cy.ts/components/progress-bar test, so I need to fix it

File: projects/kit/components/progress/progress-bar/progress-bar.component.less [hunk 2]
```diff
@@ -72,6 +72,39 @@
         height: 0.5rem;
     }
 
+    &[new] {
```
Comment (nsbarsukov) [inline projects/kit/components/progress/progress-bar/progress-bar.component.less:75]: Add `TODO`, please:

```
// TODO: drop new attribute in v4.0
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/progress/progress-bar/progress-bar.component.less [hunk 3]
```diff
@@ -72,6 +72,39 @@
...
+        &[data-size='xs'],
+        &[data-size='s'] {
```
Comment (nsbarsukov) [inline projects/kit/components/progress/progress-bar/progress-bar.component.less:77]: My mistake, I forgot that according to our design system `m`-size should be rounded too.

Could you add the following line, please ?
```suggestion
        &[data-size='s'],
        &[data-size='m'] {
```

Comment (40oleg) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less:77]: Ok, I'll fix it this evening

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**